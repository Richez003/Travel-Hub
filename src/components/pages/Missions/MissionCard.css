td {
  border: 1px solid;
}

h2 {
  margin: 1rem;
}

table {
  background: rgb(239, 239, 242);
}

th {
  border: 1px solid;
  background: #fff;
}

.progress-circle {
  width: 60px;
}

.member {
  width: 8rem;
  padding: 10px;
  background: #fff;
  text-align: center;
}

.active {
  color: #fff;
  text-align: center;
  background: blue;
  border-radius: 3px;
}

.notActive {
  color: #fff;
  text-align: center;
  background: rgb(65, 65, 66);
  border-radius: 3px;
  margin: 1rem;
}

.reserved {
  color: red;
  background: #fff;
  width: 8rem;
  padding: 8px;
  margin: 1rem;
  text-align: center;
  border: 1px solid red;
}

.joinMission {
  padding: 8px;
  width: 8rem;
  border: 1px solid rgb(43, 43, 46);
  margin: 1rem;
}

.memberRow {
  background: #fff;
}

.status {
  width: 10rem;
  text-align: center;
}
