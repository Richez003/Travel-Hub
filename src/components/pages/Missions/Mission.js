import { useSelector } from 'react-redux';
import MissionCard from './MissionCard';

const Missions = () => {
  const missions = useSelector((state) => state.missions);
  return (
    <div>
      <h2>MISSIONS</h2>
      <table>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th className="status">Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {
            missions.map((mission) => (
              <MissionCard key={mission.mission_id} mission={mission} />
            ))
          }
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
