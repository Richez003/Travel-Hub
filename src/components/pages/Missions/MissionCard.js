import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission } from '../../../redux/missions/missionsReducer';
import './MissionCard.css';

const MissionCard = ({ mission }) => {
  const dispatch = useDispatch();
  return (
    <tr className={(mission.reserved ? 'memberRow' : '')}>
      <td>{mission.mission_name}</td>
      <td>{mission.description}</td>
      <td className={(mission.reserved ? 'member' : '')}>
        {(mission.reserved ? <span className="active">Active Member</span> : <span className="notActive">Not A Member</span>)}
      </td>
      <td>
        <button
          type="button"
          onClick={() => dispatch(joinMission(mission.mission_id))}
          className={(mission.reserved ? 'reserved' : 'joinMission')}
        >
          {(mission.reserved ? 'Leave Mission' : 'Join Mission')}
        </button>
      </td>
    </tr>
  );
};

MissionCard.propTypes = {
  mission: PropTypes.shape({
    mission_name: PropTypes.string,
    mission_id: PropTypes.string,
    reserved: PropTypes.bool,
    description: PropTypes.string,
  }).isRequired,
};

export default MissionCard;
