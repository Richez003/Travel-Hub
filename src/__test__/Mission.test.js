import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Missions from '../components/pages/Missions/Mission';

describe('Missions Page Tests', () => {
  test('Missions Components Renders Correctly in DOM', () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(screen.queryByText(/Rockets/)).toBeNull();
  });
  it('correctly rendered', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
