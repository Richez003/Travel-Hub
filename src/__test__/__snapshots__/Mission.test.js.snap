// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Page Tests correctly rendered 1`] = `
<div>
  <h2>
    MISSIONS
  </h2>
  <table>
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th
          className="status"
        >
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
